pkg/scheduler/framework/plugins/clusteraffinity/plugin.go

```diff
@@ -68,6 +75,14 @@ func WithName(name string) Option {
 	}
 }
 
+// WithAzurePropertyChecker sets the Azure property checker for azure property validation.
+// This enables Azure-specific cluster requirement validation.
+func WithAzurePropertyChecker(checker *azure.PropertyChecker) Option {
```

Comment:
```suggestion
func WithPropertyChecker(checker *azure.PropertyChecker) Option {
```

in the future we may pass in a generic interface instead of a concrete azure checker.